askama_shared/src/parser.rs

```diff
@@ -63,7 +61,7 @@ pub enum Expr<'a> {
     BinOp(&'a str, Box<Expr<'a>>, Box<Expr<'a>>),
     Range(&'a str, Option<Box<Expr<'a>>>, Option<Box<Expr<'a>>>),
     Group(Box<Expr<'a>>),
-    MethodCall(Box<Expr<'a>>, &'a str, Vec<Expr<'a>>),
+    Call(Box<Expr<'a>>, Vec<Expr<'a>>),
```

Comment:
Hmm, what would happen if we use a smaller enum as the first element here, that is, one that can only be `Var`, `Method`, or `Path`?

Also it seems this commit does a bunch of refactoring of parser bits that would be nice to split out into (a) separate commit(s) if at all possible -- currently it feels like it has a lot going on.

Reply:
I don't know how I could split the second commit "Unify handling of calls" into smaller parts that still compile on its own.

Reply:
I'm it's difficult to restrict what may be on the lhs of a call: e.g. [an array](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=0213078dca6afb17cd75a240932247f6), [an addition](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1e9f6d5267ce5c76276aa7b55ff2137d).

Reply:
Fair enough!